### PR TITLE
python-gflags: 2.0 -> 3.1.1

### PR DESCRIPTION
**Pull request by benley**
###### Motivation for this change
gflags doesn't work with python 3 until version 3.1

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -11607,15 +11607,17 @@ in {
 
 
   gflags = buildPythonPackage rec {
```
Comment (Mic92) [inline pkgs/top-level/python-packages.nix]: can you also move this expression to `python-modules`?

Comment (benley) [inline pkgs/top-level/python-packages.nix]: Sure, done

**Comment by Mic92** [comment]
We have quite a few applications require gflags.
It might make sense to keep both major releases 2.x and 3.x:
https://github.com/google/python-gflags/blob/master/ChangeLog#L66

File: pkgs/development/python-modules/gflags/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+buildPythonPackage rec {
+  version = "3.1.1";
+  name = "gflags-${version}";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "python-gflags";
+    sha256 = "0qvcizlz6r4511kl4jlg6fr34y1ka956dr2jj1q0qcklr94n9zxa";
+  };
+  propagatedBuildInputs = [ six pytest ];
```
Comment (FRidh) [inline pkgs/development/python-modules/gflags/default.nix]: py.test shouldn't be propagated

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/gflags/default.nix (hunk 2 above)
Comment (Mic92) [inline pkgs/development/python-modules/gflags/default.nix]: oh, oversight.

**Pull request merged**